core/domain/subtopic_page_services_test.py

```diff
@@ -62,9 +63,26 @@ def test_get_subtopic_page_from_model(self):
         self.assertEqual(subtopic_page.to_dict(), self.subtopic_page.to_dict())
 
     def test_get_subtopic_page_by_id(self):
-        subtopic_page = subtopic_page_services.get_subtopic_page_by_id(
+        subtopic_page_1 = subtopic_page_services.get_subtopic_page_by_id(
             self.TOPIC_ID, self.subtopic_id)
-        self.assertEqual(subtopic_page.to_dict(), self.subtopic_page.to_dict())
+        self.assertEqual(
+            subtopic_page_1.to_dict(), self.subtopic_page.to_dict())
+        # When the subtopic page with the given subtopic id and topic id
+        # doesn't exist.
+        subtopic_page_2 = subtopic_page_services.get_subtopic_page_by_id(
+            'topic_id', 1, strict=False)
+        self.assertEqual(subtopic_page_2, None)
+
+    def test_get_subtopic_pages_with_ids(self):
+        subtopic_ids = [self.subtopic_id]
+        subtopic_pages = subtopic_page_services.get_subtopic_pages_with_ids(
+            self.TOPIC_ID, subtopic_ids)
+        self.assertEqual(
+            subtopic_pages[0].to_dict(), self.subtopic_page.to_dict())
+        subtopic_ids = [2]
+        with self.assertRaises(AttributeError):
```

Comment:
Why does it raise an attribute error? Shouldn't it be some variant of "element not found"?

Reply:
Looking at the below test, an appropriate exception would be base_models.BaseModel.EntityNotFoundError, could you edit the query code to raise this error? (Attribute error seems misleading for this situation).

Reply:
Hi @nithusha21, actually the function 
```
def get_subtopic_pages_with_ids(topic_id, subtopic_ids):
    """Returns a list of domain objects with given ids.

    Args:
        topic_id: str. ID of the topic that the subtopics belong to.
        subtopic_ids: list(int). The ids of the subtopics.

    Returns:
        list(SubtopicPage) or None. The list of domain objects representing the
            subtopic pages corresponding to given ids list or None if none
            exist.
    """
    subtopic_page_ids = []
    for subtopic_id in subtopic_ids:
        subtopic_page_ids.append(
            subtopic_page_domain.SubtopicPage.get_subtopic_page_id(
                topic_id, subtopic_id))
    subtopic_page_models = topic_models.SubtopicPageModel.get_multi(
        subtopic_page_ids)
    subtopic_pages = []
    for subtopic_page_model in subtopic_page_models:
        subtopic_pages.append(
            get_subtopic_page_from_model(subtopic_page_model))
    return subtopic_pages
```
Does not handle the situation if subtopic_page is ```None```, as I am sending a subtopic_ids list in which no id exists, so the subtopic_page gets None in return and thus, therefore, no object is found thus it gives Attribute Error.

Reply:
I still don't see where the attribute error is raised. This function should return an empty list I think, and you should assert that an empty list is obtained in the test. Could you specify which line fails here?

Reply:
![screenshot from 2019-03-05 23-47-52](https://user-images.githubusercontent.com/30555544/53827238-a4942400-3fa0-11e9-936b-e753500c5382.png)

Reply:
Since the subtopic_page_model is None here, so its not able to extract its id attribute.

Reply:
One option is that we can add a check after the for loop, if it is None an Exception will be raised.

Reply:
Hi @nithusha21, I have added a condition if subtopic_page_model is None it will not go for finding subtopic_page.

Reply:
That sounds better. If the model doesn't exist, it shouldn't do anything further. 👍